flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskOutputAvailabilitySampleableTaskAdapter.java

```diff
@@ -26,18 +26,18 @@
 import static java.util.Objects.requireNonNull;
 
 /**
- * Adapts {@link Task} to {@link StackTraceSampleableTask}.
+ * Adapts {@link Task} to {@link OutputAvailabilitySampleableTask}.
  */
-class TaskStackTraceSampleableTaskAdapter implements StackTraceSampleableTask {
+class TaskOutputAvailabilitySampleableTaskAdapter implements OutputAvailabilitySampleableTask {
```

Comment:
We could simplify to make `Task` class implement `BackPressureSampleableTask` directly, then to delete this class. Furthermore we could remove `#isRunning` method from the interface, because task can judge the state internally and return the boolean value via `BackPressureSampleableTask#isBackPressured`.

Reply:
There three states, e.g. back pressured, not back pressured or task not running(will throw an exception), so the ```#isRunning``` need to be reserved.